src/Application/Common/Extensions/ApplicationCacheServiceExtension.cs

```diff
@@ -0,0 +1,38 @@
+﻿using CleanArchitecture.Application.Common.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CleanArchitecture.Application.Common.Extensions
+{
+    public static class ApplicationCacheServiceExtension
+    {
+        public static T GetAndSet<T>(this IApplicationCacheService applicationCacheService, string key, Func<T> getResult, TimeSpan expireTime)
+        {
+            var data = applicationCacheService.Get<T>(key);
+
+            if (data == null)
+            {
+                data = getResult();
+                applicationCacheService.Set(key, data, expireTime);
+            }
+
+            return data;
+        }
+
+        public static T GetAndSetAsync<T>(this IApplicationCacheService applicationCacheService, string key, Func<T> getResult, TimeSpan expireTime)
```

Comment:
I can't see anything Async here except method name.

```suggestion
        public static Task<T> GetAndSetAsync<T>(this IApplicationCacheService applicationCacheService, string key, Func<Task<T>> getResult, TimeSpan expireTime)
```

Reply:
Can you check commits @enisn 5d6f5940de9969e00dedcce117b20cbf59cfe4f8 f3fddf580b797c1adbca7ce1acf813339a7f7338 ?